### DISCUSSION TITLE
Reload config

### DISCUSSION THREAD

**Discussion by fadri14** [neovim/neovim#26561] [Q&A]
Hello,
On my pc I have configured a light mode and a dark mode and so I would also like a light mode and a dark mode on neovim. I have created a separate script that changes my config file but neovim does not apply config updates in real time, so I would like to make a keyboard shortcut in neovim to reload the config but the command so ~/.config/nvim/init.lua does not reload my config.

Thanks to all those who will take the time to help me.

**Comment by rudiejd** [comment]
`set bg=light` / `set bg=dark` are likely the Vim commands you're looking for if you want to change between light and dark mode in your theme. You could call these in lua by using `vim.cmd('set bg=light)` for example, then bind some key to a function that toggles between light / dark themes.

**Comment by clason** [comment]
Reloading (all) config is not supported, since it's simply not possible in general (there is too much state that config -- including plugins -- can change).

Instead, you should write a script that only changes `bg` and reloads the colorscheme. There are already a few plugins out there that claim to sync Neovim colors to system dark mode; looking at these is your best bet.

**Comment by fadri14** [comment]
Thanks for your answers but I'd like to clarify that when I talk about light and dark themes, it's not just the background but the whole theme. That is, I use the onedark and catpuccin themes. So if I can't restart the whole config, I need to create a function that applies one of the two themes alternately? Could I use this kind of command in such a function?
require('catppuccin').load()
even if the other theme has already been applied?

**Reply by clason** [reply]
That depends on the plugin. Colorschemes _are_ meant to be reloaded (with `:colorscheme <foo>`); if the colorscheme insists to be a Lua module, then this becomes more difficult and has to be handled by the plugin itself.

**Comment by fadri14** [comment]
J'ai créé la fonction suivante mais rien ne se passe quand je l'appelle avec :lua changeTheme()…
`
function changeTheme()
    local current_theme = vim.fn.execute('colorscheme', '')

    local new_theme = ''
    if current_theme == 'onedark' then
        new_theme = 'catppuccin-latte'
    else
        new_theme = 'onedark'
    end

    vim.cmd('colorscheme ' .. new_theme)
end

`

**Reply by phanen** [reply]
Maybe you would be interested in https://github.com/4e554c4c/darkman.nvim
